### PR TITLE
Avoid empty quad in MappingInfo::reinit_cells()

### PR DESCRIPTION
**Pull request by fdrmrc**
`quadrature_vector[cell_index]` (which is an empty quadrature) was used in place of `unit_points_vector[cell_index]`.

### CODE REVIEW

**Review by luca-heltai — APPROVED** [review]
Well spotted!

File: include/deal.II/non_matching/mapping_info.h [hunk 1]
```diff
@@ -402,7 +402,7 @@ namespace NonMatching
     std::vector<Quadrature<dim>> quadrature_vector(n_cells);
     for (unsigned int cell_index = 0; cell_index < n_cells; ++cell_index)
       quadrature_vector[cell_index] =
-        Quadrature<dim>(quadrature_vector[cell_index].get_points());
+        Quadrature<dim>(unit_points_vector[cell_index]);
```
Comment (peterrum) [inline include/deal.II/non_matching/mapping_info.h:405]: @fdrmrc Great you spotted this! Could you add a test!

FYI @fdrmrc

**Review by bergbauer — APPROVED** [review]
My bad, sorry for that. We should test this code path! Thanks @fdrmrc

File: tests/non_matching/mapping_info.cc [hunk 2]
```diff
@@ -464,6 +464,87 @@ test(const bool filtered_compression)
           << dst_faces.l2_norm() - dst_faces_2.l2_norm() << std::endl;
 }
 
+
+
+void
+test_vec_of_qpoints()
```
Comment (bergbauer) [inline tests/non_matching/mapping_info.cc:470]: This should be its own test `mapping_info_02` in my opinion.

Comment (kronbichler) [reply tests/non_matching/mapping_info.cc:470]: Yes, I agree, simply move this to a new file `mapping_info_02` and create that as a new test. Thanks for adding, this is highly appreciated.

Comment (fdrmrc) [reply tests/non_matching/mapping_info.cc]: So done, thank you!

**Review by kronbichler — APPROVED** [review]

File: tests/non_matching/mapping_info_02.cc [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+  for (auto &v : src)
+    v = random_value<double>();
+  for (const auto &cell : dof_handler.active_cell_iterators())
+    {
+      cell->get_dof_values(src,
+                           solution_values_in.begin(),
+                           solution_values_in.end());
+
+      FEPointEvaluation<1, dim, dim, double> fe_point_cell(mapping_info_cell,
```
Comment (bergbauer) [inline tests/non_matching/mapping_info_02.cc:100]: I would create the `FEPointEvaluation` object outside of the cell loop.

File: tests/non_matching/mapping_info_02.cc [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                            EvaluationFlags::values |
+                              EvaluationFlags::gradients);
+          for (const auto q : fe_point.quadrature_point_indices())
+            {
+              deallog << fe_point.get_value(q) << std::endl;
+              deallog << fe_point.get_gradient(q) << std::endl;
+            }
+        };
+
+      test_fe_point_reinit_vec_pts(fe_point_cell);
```
Comment (bergbauer) [inline tests/non_matching/mapping_info_02.cc:115]: The lambda does not really make sense if you only use it for one `FEPointEvaluation` object.

**Review by bergbauer — APPROVED** [review]
Minor comments on the test, otherwise good to go!

File: tests/non_matching/mapping_info_02.cc (hunk 4 above)
Comment (fdrmrc) [reply tests/non_matching/mapping_info_02.cc]: You're right! Too much copy-paste, sorry for that

**Review by peterrum — APPROVED** [review]

File: include/deal.II/non_matching/mapping_info.h (hunk 1 above)
Comment (bangerth) [reply include/deal.II/non_matching/mapping_info.h:405]: Nice self-reference in the original code :-)

**Pull request merged**